### PR TITLE
gh-122704: Fix reference leak in `Modules/_pickle.c`

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-122704

### CODE REVIEW

File: Modules/_pickle.c [hunk 1]
```diff
@@ -1959,25 +1959,29 @@ whichmodule(PickleState *st, PyObject *global, PyObject *global_name, PyObject *
        extra parameters of __import__ to fix that. */
     module = PyImport_Import(module_name);
     if (module == NULL) {
+        Py_DECREF(module_name);
```
Comment (colesbury) [inline Modules/_pickle.c]: There's a use of `module_name` in the error message below. Maybe move the `Py_DECREF()` after the `PyErr_Format()`?

File: Modules/_pickle.c [hunk 2]
```diff
@@ -1959,25 +1959,29 @@ whichmodule(PickleState *st, PyObject *global, PyObject *global_name, PyObject *
...
     }
     if (check_dotted_path(module, global_name, dotted_path) < 0) {
+        Py_DECREF(module_name);
         Py_DECREF(module);
         return NULL;
     }
     PyObject *actual = getattribute(module, dotted_path);
     Py_DECREF(module);
     if (actual == NULL) {
+        Py_DECREF(module_name);
```
Comment (colesbury) [inline Modules/_pickle.c]: Same here

File: Modules/_pickle.c [hunk 3]
```diff
@@ -1959,25 +1959,29 @@ whichmodule(PickleState *st, PyObject *global, PyObject *global_name, PyObject *
...
         PyErr_Format(st->PicklingError,
                      "Can't pickle %R: attribute lookup %S on %S failed",
                      global, global_name, module_name);
         return NULL;
     }
     if (actual != global) {
         Py_DECREF(actual);
+        Py_DECREF(module_name);
```
Comment (colesbury) [inline Modules/_pickle.c]: Same here

File: Modules/_pickle.c (hunk 1 above)
Comment (Eclips4) [reply Modules/_pickle.c]: Indeed!

**Comment by Eclips4** [comment]
!buildbot refleaks

**Comment by Eclips4** [comment]
Hypothesis failure is unrelated, see #122686 for more details

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**